### DISCUSSION TITLE
Updated Expo Router + Expo -> Unable to resolve "framer-motion" from "node_modules/moti/build/core/index.js"

### DISCUSSION THREAD

**Discussion by johnmccants002** [nandorojo/moti#303] [Q&A] [answered]
I get this error on when trying to launch on Web: Unable to resolve "framer-motion" from "node_modules/moti/build/core/index.js"

```
"expo": "~49.0.0",
"moti": "^0.25.3",
"expo-router": "2.0.0",

```

Everything was working fine until I updated Expo Router and Expo. I ran npm i framer-motion and that didn't work either. Any ideas?

**Comment by flowgrow** [comment]
same problem here. were you able to resolve this?

**Reply by johnmccants002** [reply]
I have not. I just removed Moti for the time being. If you solve it let me know 😃

**Reply by nandorojo** [reply]
Please try this answer: https://github.com/nandorojo/moti/discussions/303#discussioncomment-7049458

**Comment by flowgrow** [comment]
I think the problem is, that expo 49 uses metro instead of webpack as its bundler, and you somehow have to configure metro to handle this. problem is, I have no idea how

**Comment by nandorojo** [comment]
are you using npm? `npm i moti —legacy-peer-deps` is what’s in the docs. does that help

**Reply by flowgrow** [reply]
yes i am using npm, yes i installed with legacy peers

**Comment by borstessi** [comment]
any news on this topic? We cannot roll out OTA Updates currently.

**Reply by nandorojo** [reply]
Are you implying that this affected your native OTA build? This is a web-only issue.

In any case, the solution is posted.

**Reply by johnmccants002** [reply]
I ended up upgrading to "moti": "^0.26.0" and adding "mjs" to my sourceExts in my metro.config.js and it worked. `resolver: {
      assetExts: assetExts.filter((ext) => ext !== "svg"),
      sourceExts: [...sourceExts, "svg", "mjs"],
    }`

**Comment by GuentonBrooks** [comment]
I decided to install framer-motion directly to see if that would help... it doesnt.
`{
"moti": "0.26.0",
"expo": "^49.0.0",
"expo-router": "^2.0.0",
"framer-motion": "10.16.4",
}`

If you're not heavily relying on the animation package, maybe indeed remove it for the time being.
If you're looking for options....
1. "React-Spring" is a decent alternative if you're building for only mobile. (that's what i used to use on bare React-Native projects). It doesn't handle multi-platform well at all though. so Either use it with React for Web (separately) or React-Native for mobile (separately).
2. Otherwise "Reanimated" is definitely tedious to work with but at least it's backed and updated together with expo so your reanimated stuff rarely breaks with updates.
3. And finally use React-Native's own "Animated" API. It's a really bad resource hog so other than fading some elements in and out here and there try not to rely heavily on it. But at least that's straight from react-native itself so it wont break unless React-Native itself does something drastic.

**Comment by nandorojo** [comment, marked as answer]
This error is due to Metro web. Please see this solution:

In your metro config file:

```typescript

const { getDefaultConfig } = require('expo/metro-config');

const config = getDefaultConfig(__dirname);

config.resolver.assetExts.push(
  // Adds support for framer-motion
  'mjs’
);

config.resolver.assetExts.push(
  'cjs’
);

module.exports = config;
```

**Reply by nandorojo** [reply]
Updated the answer for metro web / expo router 2 support

**Reply by EgorKrasno** [reply]
I am still getting `(0 , _framerMotion.usePresence) is not a function` with this fix with expo on web.

**Reply by abanobboles** [reply]
I am also

**Reply by nandorojo** [reply]
weird, maybe expo router would be a good place to ask

**Reply by nandorojo** [reply]
Hey guys, does this fix it?

```diff

// Learn more https://docs.expo.io/guides/customizing-metro
const { getDefaultConfig } = require('expo/metro-config');

/** @type {import('expo/metro-config').MetroConfig} */
const config = getDefaultConfig(__dirname);

config.transformer.getTransformOptions = async () => ({
  transform: {
    experimentalImportSupport: true,
  },
});

+config.resolver.sourceExts.push('mjs');
+config.resolver.sourceExts.push('cjs');

module.exports = config;

```

**Reply by abanobboles** [reply]
for me not, but what is working is not to use the static output for web,
![CleanShot 2023-10-09 at 20 27 00](https://github.com/nandorojo/moti/assets/9078953/13b6dccd-6d2e-4729-ac08-fbbfc7841bfc)

**Reply by ibex088** [reply]
this fixed it for me, thankyou!!

**Reply by janziemba** [reply]
It worked, thank you, Fernando!
